petab_select/plot.py

```diff
@@ -32,32 +33,185 @@
-    """Plot an UpSet plot of estimated parameters and criterion.
+class _IdentiDict(dict):
+    """Dummy dictionary that returns keys as values."""
 
-    Args:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __getitem__(self, key):
+        return key
+
+
+class PlotData:
+    """Manage data used in plots.
+
+    Attributes:
         models:
             The models.
         criterion:
             The criterion.
+        labels:
+            Model labels. Defaults to :attr:`petab_select.Model.model_label`
+            or :attr:`petab_select.Model.model_id` or
+            :attr:`petab_select.Model.hash`.
+            Keys are model hashes, values are the labels.
+        relative_criterion:
+            Whether to display relative criterion values.
+        parameter_labels:
+            Labels for parameters. Keys are parameter IDs, values are labels.
+            To use the ``parameterName`` column of your PEtab parameters table,
+            provide ``petab_problem.parameter_df["parameterName"].to_dict()``.
+        colors:
+            Colors for each model. Keys are model hashes, values are
+            matplotlib color names.
+    """
+
+    def __init__(
+        self,
+        models: Models,
+        criterion: Criterion | None = None,
+        relative_criterion: bool = False,
+        labels: dict[ModelHash, str] | None = None,
+        parameter_labels: dict[str, str] | None = None,
+        colors: dict[str, str] = None,
+    ):
+        self.models = models
+        self.criterion = criterion
+        self.relative_criterion = relative_criterion
+        self.parameter_labels = parameter_labels or _IdentiDict()
+        self.colors = colors or {}
+
+        labels = labels or {}
+        self.labels = self.get_default_labels() | labels
+        self.labels0 = copy.deepcopy(self.labels)
+
+    def get_default_labels(self) -> dict[ModelHash, str]:
+        """Get default model labels."""
+        return (
+            {
+                model.hash: model.model_label or model.model_id or model.hash
+                for model in self.models
+            }
+            | {
+                model.predecessor_model_hash: model.predecessor_model_hash
+                for model in self.models
+            }
+            | {VIRTUAL_INITIAL_MODEL.hash: "Virtual\nInitial\nModel"}
+        )
+
+    def augment_labels(
+        self,
+        criterion: bool = False,
+        added_parameters: bool = False,
+        removed_parameters: bool = False,
+    ) -> None:
+        """Add information to the plotted model labels.
+
+        N.B.: this resets any previous augmentation.
+
+        Args:
+            criterion:
+                The criterion values.
+            added_parameters:
+                The added parameters, compared to the predecessor model.
+            removed_parameters:
+                The added parameters, compared to the predecessor model.
```

Comment:
```suggestion
            criterion:
                Whether to include the criterion values.
            added_parameters:
                Whether to include the added parameters, compared to the predecessor model.
            removed_parameters:
                Whether to include the removed parameters, compared to the predecessor model.
```

Descriptions don't match the bool input type.